packages/zowe-explorer-api/src/profiles/ZoweExplorerApi.ts

```diff
@@ -648,5 +648,17 @@ export namespace ZoweExplorerApi {
          * @returns {string[]}
          */
         registeredApiTypes(): string[];
+
+        /**
+         * Register a callback function to be used when a change to the team config profile is performed.
+         * @param callback
+         */
+        registerProfileChangeCallback?(callback: (eventType: EventTypes) => Promise<void>): void;
+
+        /**
+         * Get function that is called when a change to the team config profile is performed.
+         * @returns Function
+         */
+        getProfileChangeCallback?(): (eventType: EventTypes) => Promise<void>;
```

Comment:
I'm wondering if we want to use a `vscode.EventEmitter` here instead of a callback function?

This way, developers can have more than one event listener, and we can "fire" the event emitter without needing to check if anyone is listening to the event.

I'm currently doing this in #2418 to signal events for updated USS nodes, if you're curious 😋